tests/fault_tolerance/test_vllm_health_check.py

```diff
@@ -0,0 +1,379 @@
+        pytest.fail(
+            f"Could not determine which worker handled the test request. Worker1: {worker1_received_test}, Worker2: {worker2_received_test}"
+        )
+
+
+def start_completion_request(primary_worker_name: str):
+    """
+    Start a request in a separate thread.
+
+    Args:
+        primary_worker_name: Name of the primary worker expected to handle the request
+
+    Returns:
+        tuple: (request_thread, response_queue)
+    """
+    response_queue: queue.Queue[requests.Response] = queue.Queue()
+
+    def send_formal_request():
+        response = send_completion_request(
+            "Tell me a long long long story about yourself?",
+            8000,
+            timeout=240,  # Extended timeout for long request
+        )
+        response_queue.put(response)
+
+    request_thread = threading.Thread(target=send_formal_request)
+    request_thread.start()
+
+    return request_thread, response_queue
+
+
+def validate_completion_response(
+    request_thread: threading.Thread, response_queue: queue.Queue
+):
+    """
+    Wait for and validate the completion response after worker failure.
+
+    Args:
+        request_thread: The thread running the completion request
+        response_queue: Queue containing the response from the request
+    """
+    request_thread.join(timeout=300)
+    if request_thread.is_alive():
+        pytest.fail("Request did not complete within timeout")
+
+    # Get the response
+    if response_queue.empty():
+        pytest.fail("No response received for request")
+    response = response_queue.get()
+
+    # Validate the response
+    validate_openai_response(response)
+    logger.info("✓ Request completed successfully after worker failure")
+
+
+def verify_migration_occurred(frontend_process: DynamoFrontendProcess) -> None:
+    """
+    Verify that migration occurred by checking frontend logs for stream disconnection message.
+
+    Args:
+        frontend_process: The frontend process to check logs for
+
+    Raises:
+        pytest.fail: If migration message is not found in logs
+    """
+    log_path = frontend_process._log_path
+    if not log_path or not os.path.exists(log_path):
+        pytest.fail(f"Frontend log file not found at {log_path}")
+
+    try:
+        with open(log_path, "r") as f:
+            log_content = f.read()
+            if "Stream disconnected... recreating stream..." in log_content:
+                logger.info(
+                    "✓ Migration detected: Found migration message in frontend logs"
+                )
+                return
+            else:
+                pytest.fail(
+                    "Expected migration did not occur - migration message not found in frontend logs"
+                )
+    except Exception as e:
+        pytest.fail(f"Could not read frontend log file {log_path}: {e}")
+
+
+@pytest.mark.vllm
+@pytest.mark.gpu_1
+@pytest.mark.e2e
+@pytest.mark.slow
+def test_request_migration_vllm(request, runtime_services):
```

Comment:
seems like a lot of code copied over from migration test - and also don't see a new test name here?

Reply:
there is a lot of copying - can we make separating the reusable code out to a separate component another PR?

Ah yeah, I guess I missed renaming the test itself. I'll fix that up right away.